src/services/anchor-service.ts

```diff
@@ -772,65 +772,95 @@ export class AnchorService {
+          missingRequests.length
+        } requests, but the newest request is included.  Anchoring the newest request, with CID ${newestRequest.cid.toString()}`
       )
+      candidate.setTipToAnchor(stream)
+      return
     }
 
     // If there were CIDs that we have requests for but didn't show up in the stream state that
     // we loaded from Ceramic, we can't tell if that is because those commits were rejected by
     // Ceramic's conflict resolution, or if our local Ceramic node just never heard about those
-    // commits before.  So we build a multiquery including all missing commits and send that to
-    // Ceramic, forcing it to at least consider every CID that we have a request for.
-    const queries = missingRequests.map((request) => {
-      return { streamId: CommitID.make(candidate.streamId, request.cid).toString() }
+    // commits before.  Building a multiQuery including both the base StreamID and a missing commit's CommitID
+    // will force the Ceramic node to consider that commit and sent it through conflict resolution.
+    // To be perfectly safe and consider every missing request we would need to build a multiquery with every missing
+    // request CommitID, but that is very expensive and prone to timing out.  The vast majority of the time the newest
+    // request is going to be the best one, so we only send the multiquery for the newest missing request, as an
+    // optimization.
+    logger.debug(
+      `Stream ${candidate.streamId.toString()} is missing ${
+        missingRequests.length
+      } requests from its log. The newest missing request is ${newestMissingRequest.cid.toString()} - sending multiQuery to force ceramic to load it`
+    )
+    const queries = []
+    queries.push({
+      streamId: CommitID.make(candidate.streamId, newestMissingRequest.cid).toString(),
     })
     queries.push({ streamId: candidate.streamId.baseID.toString() })
 
     // Send multiquery
     let response
     try {
       response = await ceramicService.multiQuery(queries)
+      // Get the current version of the Stream that has considered the newest missing request CID
+      stream = response[candidate.streamId.toString()]
     } catch (err) {
       logger.err(
         `Multiquery failed for stream ${candidate.streamId.toString()} with ${
           missingRequests.length
         } missing commits: ${err}`
       )
       Metrics.count(METRIC_NAMES.ERROR_MULTIQUERY, 1)
-      candidate.failAllRequests()
-      return
+      // If the multiquery fails, fall back to anchoring based on the best information available
     }
 
+    const stillMissingRequests = candidate.requests.filter((req) => {
+      const found = stream.state.log.find(({ cid }) => {
+        return cid.toString() == req.cid
+      })
+      return !found
+    })
+
     // Fail requests for tips that failed to be loaded
-    for (const request of missingRequests) {
+    for (const request of stillMissingRequests) {
```

Comment:
I believe these requests will be failed and failed due to conflict resolution (rejected). Is that the desired behavior

Reply:
yes that's the behavior.  Does the failure reason matter at this point?  I forget if we still do something based on the failure reason message